simulate/simulateIntergenic.go

```diff
@@ -12,3 +20,187 @@ func RandIntergenicSeq(GcContent float64, lenSeq int) []dna.Base {
 	}
 	return answer
 }
+
+func indelLength(lambda float64) int {
+	expFloat, _ := numbers.RandExp()
+	return int(math.Ceil(expFloat / lambda))
+}
+
+const bufferSize = 10_000_000
+
+// SimulateWithIndels takes an input fastaFile, which must contain to a single fasta entry, and simulates a mutated sequence.
+// The output sequence is provided in a multiFa alignment, aligned ot the initial sequence.
+// branchLength (a float from 0 to 1) specifies the expected value of the proportion of sites in the input sequence that will be mutated.
+// propIndel (a float from 0 to 1)specifies the expected value of the proportion of indels in the output sequence.
+// lambda specifies the rate parameter for an exponential distribution, from which simulated INDEL sizes will be sampled.
+// gcContent specifies the expected value of GC content for inserted sequences.
+// vcfOutFile specifies an optional return, which records all variants made during the simulated mutation process.
+func SimulateWithIndels(fastaFile string, branchLength float64, propIndel float64, lambda float64, gcContent float64, vcfOutFile string) []fasta.Fasta {
+	var answer = make([]fasta.Fasta, 2)
+	var emptyRoomInBuffer = bufferSize
+	var currRand, currRand2, currRand3 float64
+	var outputPos, length, indelPos, indelStartPos int
+	var newBufferRoom []dna.Base = make([]dna.Base, bufferSize)
+	var inputPos int = 0
+	var vcfOut *fileio.EasyWriter
+	var currRef, currAlt []dna.Base
+	var err error
+	var outOfChrom bool = false
+	records := fasta.Read(fastaFile)
+	if len(records) != 1 {
+		log.Fatalf("SimulateWithIndels expects a single fasta record in the input file.")
+	}
+	answer[0] = fasta.Fasta{Name: records[0].Name, Seq: make([]dna.Base, bufferSize)}
+	answer[1] = fasta.Fasta{Name: fmt.Sprintf("%v_sim", records[0].Name), Seq: make([]dna.Base, bufferSize)}
+
+	if vcfOutFile != "" {
+		vcfOut = fileio.EasyCreate(vcfOutFile)
+		_, err = fmt.Fprintf(vcfOut, "##fileformat=VCFv4.2\n#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\tFORMAT\n")
+		exception.PanicOnErr(err)
+	}
+
+	for inputPos < len(records[0].Seq) {
+		if inputPos%100_000 == 0 {
+			fmt.Printf("CurrentPos: %v.\n", inputPos)
+		}
+		currRand = rand.Float64() //this rand determines if there will be a mutation
+		if currRand < branchLength {
+			currRand2 = rand.Float64()         //this rand determines the mutation type
+			if currRand2 < (propIndel / 2.0) { //this half of INDELs will be deletions
```

Comment:
I'm trying to understand why currRand2 < or >= (propIndel / 2.0) determines if the indel will be deletion or insertion. Is it just to divide up the indel cases equally, assuming it's equally likely to get any number?